R/as_rtf.R

```diff
@@ -0,0 +1,102 @@
+#' Print a summary table using RTF
+#'
+#' Create RTF file output from \code{\link{as_table}} to summarize
+#' an object; currently only implemented for \code{\link{gsBinomialExact}}.
+#'
+#' @param x Object to be displayed on rtf.
+#' @param ... Other parameters that may be specific the object.
+#'
+#' @return RTF file for summary table.
+#'
+#' @seealso \code{vignette("binomialSPRTExample")}
+#'
+#' @details
+#' Currently only implemented for \code{\link{gsBinomialExact}} objects.
+#' Creates a table to summarize an object.
+#' For \code{\link{gsBinomialExact}}, this summarized operating characteristics
+#' across a range of effect sizes.
+#'
+#' @export
+#'
+#' @examples 
+#' zz <- gsBinomialExact(
+#'   k = 3, theta = seq(0, 1, 0.1), n.I = c(12, 24, 36),
+#'   a = c(-1, 0, 11), b = c(5, 9, 12)
+#' )
+#' zz %>% 
+#'   as_table() %>% 
+#'   as_rtf(
+#'     title = "Power/Type I error and expected sample size for a group sequential design",
+#'     file = tempfile(fileext = ".rtf")
+#'   )
+#'
+#' safety_design <- binomialSPRT(p0 = .04, p1 = .1, alpha = .04, beta = .2, minn = 4, maxn = 75)
+#' safety_power <- gsBinomialExact(
+#'   k = length(safety_design$n.I),
+#'   theta = seq(.02, .16, .02),
+#'   n.I = safety_design$n.I,
+#'   a = safety_design$lower$bound,
+#'   b = safety_design$upper$bound
+#' )
+#' safety_power %>%
+#'   as_table() %>%
+#'   as_rtf(
+#'     theta_label = "Underlying\nAE rate",
+#'     prob_decimals = 3,
+#'     bound_label = c("low rate", "high rate"),
+#'     file = tempfile(fileext = ".rtf")
+#'   )
+as_rtf <- function(x, ...) UseMethod("as_rtf")
+
+#' @rdname as_rtf
+#'
+#' @param title Table title.
+#' @param theta_label Label for theta.
+#' @param response_outcome Is outcome response rate
```

Comment:
Can we refine the explanation of `response_outcome` as `Logical values indicating if the outcome is response rate (TRUE) or failure rate (FALSE).